crates/tests/src/e2e/ibc_tests.rs

```diff
@@ -1091,6 +1185,55 @@ fn wait_for_packet_relay(
     Err(eyre!("Pending packet is still left"))
 }
 
+fn get_upgraded_height(test: &Test, min_upgrade_height: u64) -> Result<u64> {
+    std::env::set_var(ENV_VAR_CHAIN_ID, test.net.chain_id.to_string());
+    // Search the storage for the upgraded client state
+    let rpc = get_actor_rpc(test, Who::Validator(0));
+    let max_height = min_upgrade_height + 100;
+    let mut height = min_upgrade_height;
+    while height < max_height {
+        height += 1;
+        let key = upgraded_client_state_key(Height::new(0, height).unwrap());
```

Comment:
`height` never assumes the value of `min_upgrade_height` in a query. is this expected?

Reply:
Yes, it's expected, though a bit tricky because the chain isn't upgraded actually.
In this test, the upgrade height can't be specified because it doesn't know the exact height of the test wasm execution. The `min_upgrade_height` is just an estimation height to avoid searching it with all heights. This function checks the storage to look for the height.

I wanted this test with Gaia. I think it is better to upgrade a Gaia chain and then upgrade the IBC client on Namada. However, the upgrade request has been updated in the latest Gaia. When updating Gaia in Namada E2E tests, I'll change this test in the future.